packages/react-dom/src/__tests__/ReactDOMServerIntegrationHooks-test.js

```diff
@@ -867,6 +867,48 @@ describe('ReactDOMServerHooks', () => {
     });
   });
 
+  it('renders successfully after a component using hooks throws an error', () => {
+    function ThrowingComponent() {
+      const [value, dispatch] = useReducer((state, action) => {
+        return state + 1;
+      }, 0);
+
+      // throw an error if the count gets too high during the re-render phase
+      if (value >= 3) {
+        throw new Error('Error from ThrowingComponent');
+      } else {
+        // dispatch to trigger a re-render of the component
+        dispatch();
+      }
+
+      return <div>{value}</div>;
+    }
+
+    function NonThrowingComponent() {
+      const [count] = useState(0);
+      return <div>{count}</div>;
+    }
+
+    const container = document.createElement('div');
+
+    // First, render a component that will throw an error during a re-render triggered
+    // by a dispatch call.
+    try {
```

Comment:
Can you wrap this in `expect(...).toThrow()` instead so we can assert the throw happened and the message matched what we expected?

Reply:
Ah yes -- done.